scripts/check_phase42_boundary_correspondence.py

```
@@ -0,0 +1,1074 @@
+    hasher.update(value.to_bytes(16, "little"))
+
+
+def phase30_hash_part(hasher: Any, value: str | bytes) -> None:
+    raw = value if isinstance(value, bytes) else value.encode()
+    hasher.update(len(raw).to_bytes(8, "little"))
+    hasher.update(raw)
+
+
+def commit_phase12_layout(layout: dict[str, Any]) -> str:
+    if set(layout) != {"layout_version", "rolling_kv_pairs", "pair_width"}:
+        raise Phase42Error("Phase30 layout: unexpected fields")
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)